project_milestone_status/models/project.py

```diff
@@ -0,0 +1,109 @@
+import ast
+import json
+
+from odoo import _, models
+
+
+class Project(models.Model):
+    _inherit = "project.project"
+
+    def _get_execution(self):
+        fold_stage_ids = self.env["project.task.type"].search([("fold", "=", True)])
```

Comment:
All those searches should be done once only. Then, filter results.

```suggestion
        all_tasks = self.env["project.task"].search(
            [
                ("project_id", "=", self.id),
            ]
        )
        executed_tasks = all_tasks.filtered("fold_stage_ids.fold")
```

Reply:
and the better is to use directly `self.tasks_ids`

Reply:
> and the better is to use directly `self.tasks_ids`

The problem with `self.task_ids` is that it doesn't return all the tasks, only the open ones.